Mastodon/Protocol/Provider/DataSourceFacade+Notifications.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright © 2024 Mastodon gGmbH. All rights reserved.
+
+import Foundation
+import MastodonCore
+import MastodonSDK
+
+extension DataSourceFacade {
+    @MainActor
+    static func coordinateToNotificationRequests(
+        provider: DataSourceProvider & AuthContextProvider
+    ) async {
+        provider.coordinator.showLoading()
+
+        do {
+            let notificationRequests = try await provider.context.apiService.notificationRequests(authenticationBox: provider.authContext.mastodonAuthenticationBox).value
+            let viewModel = NotificationRequestsViewModel(appContext: provider.context, authContext: provider.authContext, coordinator: provider.coordinator, requests: notificationRequests)
+
+            provider.coordinator.hideLoading()
+
+            let transition: SceneCoordinator.Transition
+
+            if provider.traitCollection.userInterfaceIdiom == .phone {
+                transition = .show
+            } else {
+                transition = .modal(animated: true)
+            }
+
+            provider.coordinator.present(scene: .notificationRequests(viewModel: viewModel), transition: transition)
+        } catch {
+            //TODO: Error Handling
```

Comment:
@zeitschlag is there anything we should add here for error handling?

Reply:
We don't handle errors elsewhere (i.e. show an alert or something) IIRC, so I don't think so?